python/openchange/mailbox.py

```diff
@@ -281,24 +288,93 @@ def set_message_ChangeNumber(self, server, ChangeNumber):
             self.ldb.transaction_commit()
 
 
-class OpenChangeDBWithMysqlBackend(object):
-    """The OpenChange database."""
+class MysqlBackendMixin(object):
+    """Mixin for common methods for MySQL backends.
+
+    It expects the following attributes when calling its methods:
+
+    * url string for MySQL connection string.
+    * migration_app string for migration-related methods."""
+
+    def _connect_to_mysql(self):
+        host, user, passwd, self.db_name, port = self._parse_mysql_url()
+        self.db = MySQLdb.connect(host=host, user=user, passwd=passwd, port=port)
 
-    OPENCHANGEDB_SCHEMA = 'openchangedb/openchangedb_schema.sql'
+    def _parse_mysql_url(self):
+        # self.url should be mysql://user[:passwd]@some_host[:port]/some_db_name
+        if not self.url.startswith('mysql://'):
+            raise ValueError("Bad connection string for mysql: invalid schema")
+
+        if '@' not in self.url:
+            raise ValueError("Bad connection string for mysql: expected format "
+                             "mysql://user[:passwd]@host/db_name")
+
+        if '/' not in self.url:
+            raise ValueError("Bad connection string for mysql: expected format "
+                             "mysql://user[:passwd]@host/db_name")
```

Comment:
Why don't you use match return value to raise this kind of errors? if conn_string_match is None: raise ValueError